plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/delete/DeletionVectors.java

```diff
@@ -0,0 +1,135 @@
+            throw new TrinoException(NOT_SUPPORTED, "Unsupported storage type for deletion vector: " + deletionVector.storageType());
+        }
+        throw new IllegalArgumentException("Unexpected storage type: " + deletionVector.storageType());
+    }
+
+    public static String toFileName(String pathOrInlineDv)
+    {
+        int randomPrefixLength = pathOrInlineDv.length() - Base85Codec.ENCODED_UUID_LENGTH;
+        String randomPrefix = pathOrInlineDv.substring(0, randomPrefixLength);
+        checkArgument(ALPHANUMERIC.matchesAllOf(randomPrefix), "Random prefix must be alphanumeric: %s", randomPrefix);
+        String prefix = randomPrefix.isEmpty() ? "" : randomPrefix + "/";
+        String encodedUuid = pathOrInlineDv.substring(randomPrefixLength);
+        UUID uuid = decodeUuid(encodedUuid);
+        return "%sdeletion_vector_%s.bin".formatted(prefix, uuid);
+    }
+
+    public static ByteBuffer readDeletionVector(TrinoInputFile inputFile, int offset, int expectedSize)
+            throws IOException
+    {
+        byte[] bytes = new byte[expectedSize];
+        try (DataInputStream inputStream = new DataInputStream(inputFile.newStream())) {
+            checkState(inputStream.skip(offset) == offset);
+            int actualSize = inputStream.readInt();
+            if (actualSize != expectedSize) {
+                throw new TrinoException(DELTA_LAKE_INVALID_SCHEMA, "The size of deletion vector %s expects %s but got %s".formatted(inputFile.location(), expectedSize, actualSize));
+            }
+            inputStream.readFully(bytes);
+        }
+        return ByteBuffer.wrap(bytes).order(LITTLE_ENDIAN);
+    }
+
+    private static Roaring64NavigableMap deserializeDeletionVectors(ByteBuffer buffer)
+            throws IOException
+    {
+        checkArgument(buffer.order() == LITTLE_ENDIAN, "Byte order must be little endian: %s", buffer.order());
+        int magicNumber = buffer.getInt();
+        if (magicNumber == PORTABLE_ROARING_BITMAP_MAGIC_NUMBER) {
+            int size = toIntExact(buffer.getLong());
+            Roaring64NavigableMap bitmaps = new Roaring64NavigableMap();
+            bitmaps.add();
+
+            for (int i = 0; i < size; i++) {
+                int key = buffer.getInt();
+                checkArgument(key >= 0);
+
+                RoaringBitmap bitmap = new RoaringBitmap();
+                bitmap.deserialize(buffer);
+                bitmap.stream().forEach(bitmaps::add);
+
+                // there seems to be no better way to ask how many bytes bitmap.deserialize has read
+                int consumedBytes = bitmap.serializedSizeInBytes();
+                buffer.position(buffer.position() + consumedBytes);
+            }
+            return bitmaps;
+        }
+        throw new IllegalArgumentException("Unsupported magic number: " + magicNumber);
+    }
+
+    public static UUID decodeUuid(String encoded)
+    {
+        ByteBuffer buffer = decodeBlocks(encoded);
```

Comment:
do not import statically (unless the method name is something like "decodeBase85")